### ISSUE TITLE
Default value for SafeMaskMaker and etrue in the HLI

### ISSUE THREAD

**Issue by JouvinLea** [gammapy/gammapy#2688]
Hi all,
Last week, I encountered some issues running 1D analysis from the HLI for two reasons:
- 1) The binning in etrue was not propagated from the config file to the spectrum extraction where you adopted some hardcoded value. This is now fixed by @registerrier (https://github.com/gammapy/gammapy/pull/2673)

- 2) In the HLI, you are also adopting a default method to compute the safemaskmaker for the 1D extraction. I personally don't use this functionality for MAGIC data in my scripts not using the HLI. So I need a way to not compute it from the HLI as well.
I started coded something to get rid of the issue but I know it is not the right way to go. Maybe we should have also a larger discussion on what should be use as default value for etrue or for this safeMakerMask.
Personally for the safeMakerMask I will add a method in the class that if it is equal to None it doesn’t do anything and I will use that as default for the HLI. We can add a member, methods_safe_mask_maker, in the class DatasetsConfig of the HLI.
Then I will make it very clear in the notebook of the HLI for 1D analysis how to modify this safemaskmethod. Also we add to add a notebook for the HLI for 1D analysis.

What do you think? Any opinion on that?

**Comment by adonath** [comment]
Thanks @JouvinLea, you proposal goes in the same direction that @luca-giunti suggest.

The solution is just to expose all configuration from the `SafeMaskMaker` and if no config is present, no safe mask is computed at all.

In the meantime you can always "delete" the safe mask from the dataset after data reduction by setting it to `None` like so:
```python3
for dataset in analysis.datasets:
    dataset.mask_safe = None
```
The safe mask is only applied to the data during stacking. For joint analyses you can always modify it any time later. If needed you can also re-stack.

**Comment by adonath** [comment]
Fixed in #2747 and #2673.
